bundles/AdminBundle/Controller/Admin/DataObject/ClassController.php

```diff
@@ -1745,6 +1745,45 @@ public function getBrickUsagesAction(Request $request)
         return $this->adminJson($result);
     }
 
+    /**
+     * @Route("/get-icons", methods={"GET"})
+     *
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function getIconsAction(Request $request)
+    {
+        $classId = $request->get('classId');
+
+        $path = "/bundles/pimcoreadmin/img/object-icons";
+        $icons = scandir(PIMCORE_WEB_ROOT . $path);
+
+        array_shift($icons);
+        array_shift($icons);
```

Comment:
Is this to remove the `.` and `..` directories? Wouldn't it be nicer to use [Symfony Finder](https://symfony.com/doc/current/components/finder.html) and filter by file name?

Reply:
or just use rscandir helper https://github.com/pimcore/pimcore/blob/4958e801727dc5541688cd6d83f9aa7c4be69d04/bundles/AdminBundle/Resources/views/Admin/Misc/iconList.html.php#L7

Reply:
thx for your feedback please see the latest commit